src/main/java/gregtech/common/metatileentities/multi/MetaTileEntityLargeBoiler.java

```diff
@@ -409,14 +405,15 @@ protected BlockPattern createStructurePattern() {
                 .aisle("XXX", "CCC", "CCC", "CCC")
                 .aisle("XXX", "CPC", "CPC", "CCC")
                 .aisle("XXX", "CSC", "CCC", "CCC")
-                .setAmountAtLeast('X', 4)
-                .setAmountAtLeast('C', 20)
                 .where('S', selfPredicate())
-                .where('P', statePredicate(boilerType.pipeState))
-                .where('X', state -> statePredicate(boilerType.fireboxState)
-                        .or(abilityPartPredicate(MultiblockAbility.IMPORT_FLUIDS, MultiblockAbility.IMPORT_ITEMS, MultiblockAbility.MUFFLER_HATCH)).test(state))
-                .where('C', statePredicate(boilerType.casingState).or(abilityPartPredicate(
-                        MultiblockAbility.EXPORT_FLUIDS, MultiblockAbility.MAINTENANCE_HATCH)))
+                .where('P', states(boilerType.pipeState))
+                .where('X', states(boilerType.fireboxState).setMinGlobalLimited(4)
+                        .or(abilities(MultiblockAbility.IMPORT_FLUIDS).setMinGlobalLimited(1))
+                        .or(abilities(MultiblockAbility.IMPORT_ITEMS).setMinGlobalLimited(1))
+                        .or(abilities(MultiblockAbility.MUFFLER_HATCH).setMinGlobalLimited(1).setMaxGlobalLimited(1)))
```

Comment:
The Muffler hatch should be in the casings, not the firebox casings

Reply:
well, it used to be there. should be move to the casings?